scipy/sparse/linalg/_isolve/symmlq.py

```diff
@@ -0,0 +1,239 @@
+import numpy as np
+from .utils import make_system
+
+
+__all__ = ['symmlq']
+
+
+def symmlq(A, b, x0=None, tol=1e-5, maxiter=None, M=None, callback=None,
+           atol=None, haptol=1e-18, show=False):
+    """
+    Use Symmetric LQ iteration to solve ``Ax = b``.
+
+    Parameters
+    ----------
+    A : {sparse matrix, ndarray, LinearOperator}
+        The real or complex N-by-N matrix of the linear system.
+        Alternatively, `A` can be a linear operator which can
+        produce ``Ax`` using, e.g.,
+        `scipy.sparse.linalg.LinearOperator`.
+    b : {ndarray}
+        Right hand side of the linear system. Has shape (N,) or (N, 1).
+    x0 : {ndarray}
+        Starting guess for the solution.
+    tol, atol : float, optional
+        Tolerances for convergence, ``norm(b - Ax) < max(tol*norm(b), atol)``.
+        The default for `tol` is 1.0e-5.
+        The default for `atol` is ``tol * norm(b)``.
+
+        .. warning::
+
+           The default value for `atol` will be changed in a future release.
+           For future compatibility, specify `atol` explicitly.
+    haptol : float, optional
+        Tolerance for happy breakdown, ``abs(np.inner(r.conjugate(), Mr)) <
```

Comment:
What is happy breakdown? This shouldn't be exposed to user here in my opinion.

Reply:
> What is happy breakdown? 

The "Happy Breakdown" mathematically refers to $r^*Mr = 0$ (that is the obtained solution $x$ is exact), where $r:=b-Ax$, $*$ is Hermitian conjugate, and M is a positive-definite preconditioner used by Krylov method.

> This shouldn't be exposed to user here in my opinion.

You are right, it shouldn't be exposed to user. I'll fix it, thanks.